projects/extension/src/background/AppMediator.ts

```diff
@@ -174,10 +209,33 @@ export class AppMediator {
     this.#handleDisconnect(true);
   }
 
+  #sendUnsubscribe = (sub: SubscriptionMapping): void => {
+    // use one higher than we've seen before from the UApp.  The UApp is now
+    // disconnnecting so this won't ever be reused as we no longer
+    // accept incoming RPC send requests
+    const appID = ++this.highestUAppRequestId;
+    const unsubRequest = {
+      id: appID,
+      jsonrpc: '2.0',
+      method: sub.method,
+      params: [ sub.subID ]
+    }
+
+    // send the unsubscribe message
+    const smoldotID = this.#manager.sendRpcMessageTo(this.#smoldotName as string,  unsubRequest);
+    // track the request so we know when its completed
+    this.requests.push({ appID, smoldotID });
+  };
+
   #handleDisconnect = (notify: boolean): void => {
-    console.log(`Disconnecting and notify is set to ${notify.toString()}`);
+    if (this.#state === 'disconnecting' || this.#state === 'disconnected') {
+      throw new Error('Cannot disconnect - already disconnecting / disconnected');
```

Comment:
Im not sure if this should throw an error or return an error message in a specific format for uApp to handle (This is not an exception error but a "flow/time" error)

Reply:
I did it this way because it's "programmer error"  ... this happens if a the port disconnects twice I.E. it should not ever happen.. a chrome.runtime.Port should never trigger a disconnect event twice.  If it does we really want to know about it because it means my assumptions about ports are all wrong! Hence an error.